front/lib/plans/usage/types.ts

```diff
@@ -0,0 +1,53 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type Stripe from "stripe";
+
+// This is the key used in Stripe's metadata to indicate that this is a usage-based price.
+export const REPORT_USAGE_METADATA_KEY = "REPORT_USAGE";
+
+export class InvalidReportUsageError extends Error {}
+
+export const SUPPORTED_REPORT_USAGE = ["MAU_1", "MAU_5", "MAU_10"] as const;
+export type SupportedReportUsage = (typeof SUPPORTED_REPORT_USAGE)[number];
+
+export function isSupportedReportUsage(
+  usage: string | undefined
+): usage is SupportedReportUsage {
+  return SUPPORTED_REPORT_USAGE.includes(usage as SupportedReportUsage);
+}
+
+/**
+ * Monthly active users logic.
+ */
+
+export type MauReportUsageType = `MAU_${number}`;
```

Comment:
what about using  SupportedReportUsage type instead of this one ? this way we never risk the number being wrong (in the forseeable future we'll keep updated SupportedUsageType anyways)

Reply:
This one is here to enforce the type of usage for the function `reportMonthlyActiveUsers`. Since `SupportedReportUsage` will contain other types in the future, it's to ensure that the underlying implementation are properly done in the switch/case block.